src/applications/vaos/referral-appointments/tests/hooks/useGetReferralById/useGetReferralById.unit.spec.js

```diff
@@ -1,46 +1,117 @@
+import { FETCH_STATUS } from '../../../../utils/constants';
 
 import TestComponent from './TestComponent';
 
 describe('Community Care Referrals', () => {
-  describe('useGerReferralById hook', () => {
+  describe('useGetReferralById hook', () => {
+    const now = new Date();
     const sandbox = sinon.createSandbox();
-    let store;
-    const api = { fetchReferral: getPatientReferralById };
     beforeEach(() => {
       global.XMLHttpRequest = sinon.useFakeXMLHttpRequest();
-      const middleware = [];
-      const mockStore = configureStore(middleware);
-      const initState = {
+    });
+    afterEach(() => {
+      sandbox.restore();
+    });
+    it('Loads test component with hook and fetches referral when state is empty', () => {
+      sandbox
+        .stub(getPatientReferralByIdModule, 'getPatientReferralById')
+        .resolves(createReferral(now, '111'));
+      const initialState = {
         featureToggles: {
           vaOnlineSchedulingCCDirectScheduling: true,
         },
         referral: {
           facility: null,
           referrals: [],
-          referralFetchStatus: 'notStarted',
+          referralFetchStatus: FETCH_STATUS.notStarted,
         },
       };
-      store = mockStore(initState);
-      sandbox.stub(api, 'fetchReferral').resolves({});
+      const screen = renderWithStoreAndRouter(<TestComponent />, {
+        initialState,
+      });
+      expect(screen.getByText(/Test component/i)).to.exist;
+      sandbox.assert.calledOnce(
+        getPatientReferralByIdModule.getPatientReferralById,
+      );
     });
-    afterEach(() => {
-      sandbox.restore();
+    it('Returns the referral from store if exists without calling endpoint', () => {
+      sandbox
+        .stub(getPatientReferralByIdModule, 'getPatientReferralById')
+        .resolves(createReferral(now, '111'));
+      const initialState = {
+        featureToggles: {
+          vaOnlineSchedulingCCDirectScheduling: true,
+        },
+        referral: {
+          facility: null,
+          referrals: [
+            createReferral(now, 'add2f0f4-a1ea-4dea-a504-a54ab57c6800'),
+          ],
+          referralFetchStatus: FETCH_STATUS.notStarted,
+        },
+      };
+      const screen = renderWithStoreAndRouter(<TestComponent />, {
+        initialState,
+      });
+      expect(screen.getByText('add2f0f4-a1ea-4dea-a504-a54ab57c6800')).to.exist;
+      sandbox.assert.notCalled(
+        getPatientReferralByIdModule.getPatientReferralById,
+      );
+    });
+    it('Returns the referral not found when referral not in store', () => {
+      sandbox
+        .stub(getPatientReferralByIdModule, 'getPatientReferralById')
+        .resolves(createReferral(now, '111'));
+      const initialState = {
+        featureToggles: {
+          vaOnlineSchedulingCCDirectScheduling: true,
+        },
+        referral: {
+          facility: null,
+          referrals: [
+            createReferral(now, 'add2f0f4-a1ea-4dea-a504-a54ab57c6801'),
```

Comment:
Shouldn't the referral array be empty for this test?

Reply:
This is a test to return `referralNotFound` as true, when it can't find the specific one in the store. It needs referrals present to look through and not call the endpoint

Reply:
the referral the component is asking for is `add2f0f4-a1ea-4dea-a504-a54ab57c6800` and the only one present is `add2f0f4-a1ea-4dea-a504-a54ab57c6801`